src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -73,45 +73,73 @@ public override BoundNode VisitConversion(BoundConversion node)
             return result;
         }
 
-        private BoundExpression CreateUTF8ByteRepresentation(SyntaxNode resultSyntax, SyntaxNode valueSyntax, string value, ArrayTypeSymbol byteArray)
+        public override BoundNode VisitUTF8String(BoundUTF8String node)
         {
-            Debug.Assert(byteArray.IsSZArray);
-            Debug.Assert(byteArray.ElementType.SpecialType == SpecialType.System_Byte);
+            Debug.Assert(node.Type.OriginalDefinition.Equals(_compilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T), TypeCompareKind.AllIgnoreOptions));
+            var byteType = ((NamedTypeSymbol)node.Type).TypeArgumentsWithAnnotationsNoUseSiteDiagnostics.Single().Type;
+            Debug.Assert(byteType.SpecialType == SpecialType.System_Byte);
+
+            var save_Syntax = _factory.Syntax;
+            _factory.Syntax = node.Syntax;
 
-            var utf8 = new System.Text.UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
-            byte[] bytes;
+            int length;
+            BoundExpression utf8Bytes = createUTF8ByteRepresentation(node.Syntax, node.Syntax, node.Value, ArrayTypeSymbol.CreateSZArray(_compilation.Assembly, TypeWithAnnotations.Create(byteType)), out length);
+            BoundNode result;
 
-            try
+            if (!TryGetWellKnownTypeMember<MethodSymbol>(node.Syntax, WellKnownMember.System_ReadOnlySpan_T__ctor_Array_Start_Length, out MethodSymbol ctor))
             {
-                bytes = utf8.GetBytes(value);
+                result = BadExpression(node.Syntax, node.Type, ImmutableArray<BoundExpression>.Empty);
             }
-            catch (Exception ex)
+            else
             {
-                _diagnostics.Add(
-                    ErrorCode.ERR_CannotBeConvertedToUTF8,
-                    valueSyntax.Location,
-                    ex.Message);
-
-                return BadExpression(resultSyntax, byteArray, ImmutableArray<BoundExpression>.Empty);
+                result = new BoundObjectCreationExpression(node.Syntax, ctor.AsMember((NamedTypeSymbol)node.Type), utf8Bytes, _factory.Literal(0), _factory.Literal(length));
             }
 
-            var builder = ArrayBuilder<BoundExpression>.GetInstance(bytes.Length);
-            foreach (byte b in bytes)
+            _factory.Syntax = save_Syntax;
+
+            return result;
+
+            BoundExpression createUTF8ByteRepresentation(SyntaxNode resultSyntax, SyntaxNode valueSyntax, string value, ArrayTypeSymbol byteArray, out int length)
```

Comment:
nit: should 'resultSyntax' and 'valueSyntax' be replaced with a single argument here?